From 60753249c3e4247a63a394766d1b9e35b8d04fa7 Mon Sep 17 00:00:00 2001
From: Pavel Valena <pvalena@redhat.com>
Date: Tue, 25 Apr 2023 14:56:59 +0200
Subject: [PATCH 1/6] feat(hwdb): add hwdb module to install hwdb.bin on demand

Module to install hwdb.bin. Further extensions might make only selected
part of hwdb installable to save space. The module is not included by default.

Including the module adds 2MB of compressed data (on Fedora, the file has 12MB).

Installing hwdb.bin is needed in case of custom HW like a keyboard/mouse, or various interfaces.

Original PR: https://github.com/dracutdevs/dracut/pull/1681
---
 modules.d/95hwdb/module-setup.sh | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/modules.d/95hwdb/module-setup.sh b/modules.d/95hwdb/module-setup.sh
index 19112be5..df4a98da 100644
--- a/modules.d/95hwdb/module-setup.sh
+++ b/modules.d/95hwdb/module-setup.sh
@@ -2,15 +2,17 @@
 # This file is part of dracut.
 # SPDX-License-Identifier: GPL-2.0-or-later
 
+check() {
+    return 255
+}
+
 # called by dracut
 install() {
-    inst_multiple -o \
-        /etc/udev/udev.hwdb \
-        "${udevdir}"/hwdb.bin
+    # Follow the same priority as `systemd-hwdb`; `/etc` is the default
+    # and `/usr/lib` an alternative location.
+    inst_multiple "${udevconfdir}"/hwdb.bin
 
-    # Install the hosts local user configurations if enabled.
     if [[ $hostonly ]]; then
-        inst_multiple -H -o \
-            "$udevconfdir"/hwdb.bin
+        inst_multiple -H -o "${udevdir}"/hwdb.bin
     fi
 }
-- 
2.42.0

